cmd/update-aka-ms/update-aka-ms.go

```diff
@@ -0,0 +1,237 @@
+		"If specified, must match the build asset JSON's version or this command will fail.\n"+
+		"The string 'nil' is treated as the same as not setting the value, for use in CI.\n"+
+		"Optionally use this to validate expectations.")
+
+func main() {
+	help := flag.Bool("h", false, "Print this help message.")
+	buildAssetJSON := flag.String("build-asset-json", "", "[Required] The path of a build asset JSON file describing the Go build to update to.")
+
+	flag.Usage = func() {
+		fmt.Fprintf(flag.CommandLine.Output(), "\nUsage:\n")
+		flag.PrintDefaults()
+		fmt.Fprintf(flag.CommandLine.Output(), "%s\n\n", description)
+	}
+	flag.Parse()
+
+	if *help {
+		flag.Usage()
+		os.Exit(0)
+	}
+
+	if *buildAssetJSON == "" {
+		flag.Usage()
+		log.Fatal("No build asset JSON specified.\n")
+	}
+	if *validateVersionFlag == "nil" {
+		*validateVersionFlag = ""
+	}
+
+	if err := createAkaMSLinks(*buildAssetJSON); err != nil {
+		log.Fatalf("error: %v\n", err)
+	}
+
+	log.Println("\nSuccess.")
+}
+
+func createAkaMSLinks(assetFilePath string) error {
+	wd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	akaMSDir := filepath.Join(wd, "eng", "artifacts", "akams")
+	if err := os.MkdirAll(akaMSDir, os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+
+	var b buildassets.BuildAssets
+	if err := buildmodel.ReadJSONFile(assetFilePath, &b); err != nil {
+		return err
+	}
+
+	if *validateVersionFlag != "" {
+		assetVersion := b.GetGoVersion().Full()
+		inputVersion := goversion.New(*validateVersionFlag).Full()
+		if assetVersion != inputVersion {
+			return fmt.Errorf("build asset JSON version %q doesn't match input version %q", assetVersion, inputVersion)
+		}
+	}
+
+	linkPairs, err := createLinkPairs(b)
+	if err != nil {
+		return err
+	}
+
+	propsFileContent, err := getPropsFileContent(linkPairs)
+	if err != nil {
+		return err
+	}
+
+	projectPath := filepath.Join(wd, "eng", "publishing", "UpdateAkaMSLinks", "UpdateAkaMSLinks.csproj")
+	propsPath := filepath.Join(akaMSDir, "AkaMSLinks.props")
+	if err := os.WriteFile(propsPath, []byte(propsFileContent), 0666); err != nil {
+		return err
+	}
+
+	log.Printf("---- File content for generated file %v\n%v\n", propsPath, propsFileContent)
+
+	cmd := exec.Command(
+		"dotnet", "build", projectPath,
+		fmt.Sprintf("/p:LinkItemPropsFile=\"%v\"", propsPath))
+	// Pass any additional args through. Likely /p:Key=Value and /bl:Something.binlog
+	cmd.Args = append(cmd.Args, flag.Args()...)
+	return executil.Run(cmd)
+}
+
+type akaMSLinkPair struct {
+	short, target string
+}
+
+func (p *akaMSLinkPair) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	s := xml.StartElement{
+		Name: xml.Name{Local: "AkaMSLink"},
+		Attr: []xml.Attr{
+			{Name: xml.Name{Local: "Include"}, Value: p.short},
+			{Name: xml.Name{Local: "TargetUrl"}, Value: p.target},
+		},
+	}
+	err := e.EncodeToken(s)
+	if err != nil {
+		return err
+	}
+	return e.EncodeToken(s.End())
+}
```

Comment:
You can use the `xml` tag to get the same encoding result without having to implement `MarshalXML`: https://play.golang.com/p/VzVqUbLdgVr